guide/popular-topics/errors.md

```diff
@@ -0,0 +1,155 @@
+# Errors
+
+There is no doubt that you have encountered errors while making bots. While errors are very useful at warning you of what is going wrong, many people are stumped by them and how to track them down and fix them, but don't worry, we have you covered. This section will be all about how to diagnose errors, how to identify where they are coming from, and how to fix them.
+
+## Types of Errors
+
+### API Errors
+
+API Errors or DiscordAPIErrors are errors which are thrown by the Discord API when an invalid request is carried out. API Errors can be mostly diagnosed using the message that is given. They can also be further examined by seeing the http method and path used, we will explore tracking these errors down in the next section.
+
+Example: `DiscordAPIError: Cannot send an empty message`
+
+### Discord.js Errors
+
+Discord.js Errors are errors which are thrown by the library itself, they can usually be easily tracked down using the stacktrace and error message.
+
+Example: `Error: Members didn't arrive in time.`
+
+### JS Errors
+
+JS Errors are simple errors which can be thrown by node itself, or by discord.js. These types of errors can easily be fixed by looking at the type of error, and the stacktrace. You can find a full list of types [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error) And a list of common js errors [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Errors).
+
+Example: `ReferenceError: "x" is not defined`
+
+### Websocket and Network Errors
+
+Websocket and Network errors are common system errors thrown by Node in response to something wrong with the websocket connection. Unfortunately, these errors do not have a concrete solution and can be (usually) fixed by getting a better, and stronger connection. However, if you handle these errors correctly, discord.js will attempt to reconnect to the websocket automatically.
+
+You can handle these errors by adding an error listener as shown below.
```

Comment:
"Handle" is a bit of a *stretch* — you're preventing the process from crashing, sure, but you aren't really... doing anything with the error; just logging it. Maybe a slight reword would work

Reply:
i agree, i will reword it to something along the lines of `you can prevent these errors from crashing your process with`..., would that be good?

Reply:
Yep yep, sounds good! Then aaagain, console.log is just a fancier no-op to prevent the process from crashing, you should handle those errors regardless

Reply:
yeah, its just an issue of "the only real solution is get a better connection" to handle these errors